internal/executor/service/job_lease.go

```diff
@@ -237,3 +242,11 @@ func chunkPods(pods []*v1.Pod, size int) [][]*v1.Pod {
 	}
 	return chunks
 }
+
+func evictRetriedJobs(retryCache RetryCache, jobIds []string) {
+	for _, jobId := range jobIds {
+		if retryCache.GetNumberOfRetryAttempts(jobId) > 0 {
+			retryCache.Evict(jobId)
```

Comment:
Since `delete(map, key)` is safe no matter what, do we actually need this check here?

Reply:
The only reason I put the check there is because `GetNumberOfRetryAttempts` only requires a `RLock`, whilst `Evict` uses a  write `Lock` (i.e. to avoid having unnecessary writer locks). Do you think this is unnecessary?